builder/test/softmax.cc

```diff
@@ -139,9 +145,19 @@ TEST_P(softmax, pipeline) {
       {{exp_in, {c, b}}, {sum_exp_in, {b}}}, call_stmt::attributes{.name = "exp_in"});
   func pass3 = func::make(normalize, {{exp_in, {all_c, point(b)}}, {sum_exp_in, {point(b)}}}, {{softmax_out, {c, b}}},
       call_stmt::attributes{.name = "normalize"});
-  // Add a trivial consumer so we can have an inner loop here too.
-  func pass4 = func::make(
-      add_1<float>, {{softmax_out, {point(c), point(b)}}}, {{out, {c, b}}}, call_stmt::attributes{.name = "consumer"});
+
+  func pass4, copy;
+
+  if (copy_in_the_end) {
+    // Add a trivial consumer so we can have an inner loop here too.
+    pass4 = func::make(add_1<float>, {{softmax_out, {point(c), point(b)}}}, {{add_out, {c, b}}},
```

Comment:
I think this would be better like:
```
    // Add a trivial consumer so we can have an inner loop here too.
    func pass4 = func::make(add_1<float>, {{softmax_out, {point(c), point(b)}}}, {{add_out, {c, b}}}, call_stmt::attributes{.name = "consumer"});
    func copy;
    if (copy_at_the_end) {
      copy = func::make_copy({add_out, {point(c), point(b)}}, {out, {c, b}});
    } else {
      out = add_out;
    }
```

Reply:
Done.

Reply:
Surprisingly, this doesn't work when copy_at_the_end is false and fails with `value` undefined assert. I've put a condition into pass4 definition for now, but probably is worth investigating.